tensorflow/compiler/xla/service/pattern_matcher.h

```diff
@@ -1266,6 +1266,56 @@ class HloInstructionPatternOpcodeImpl {
   bool invert_;
 };
 
+// An HloInstructionPattern implementation that optionally matches a unary
+// operand with a given opcode before matching a given pattern.
+template <typename OperandType, typename OperandImpl>
+class HloInstructionPatternUnaryOptionalOpImpl {
+ public:
+  explicit HloInstructionPatternUnaryOptionalOpImpl(
+      absl::Span<const HloOpcode> opcodes,
+      const HloInstructionPattern<OperandType, OperandImpl>& operand)
+      : opcodes_(opcodes), operand_(operand) {}
+
+  bool Match(::xla::HloInstruction* inst, MatchOption option) const {
+    // In the transparent case, the instruction matches the given operand
+    // pattern.
+    if (operand_.Match(inst, option)) {
+      return true;
+    }
+    // Alternatively, the opcode of the instruction matches one of the entries
+    // of opcodes_.
+    if (!c_linear_search(opcodes_, inst->opcode())) {
+      EXPLAIN << " HloInstruction also doesn't have any of opcodes {"
+              << absl::StrJoin(opcodes_, ", ", opcode_to_string) << "}.";
+      return false;
+    }
+    // Additionally, the operand of the instruction must match the given operand
+    // pattern.
+    if (!operand_.Match(HloOperand(inst, 0), option)) {
+      EXPLAIN
+          << " Matched HloInstruction " << HloOpcodeString(inst->opcode())
+          << ", but its operand differs from the prescribed operand pattern.";
```

Comment:
I don't particularly care that we unit test this branch, but I would like to at least see what the error looks like in context, since I had opinions on the above error.  Probably the easiest way to show what it looks like would in fact be to write a unit test.

Reply:
In this case, the explanation by the negative `operand_.Match()` precedes anything introduced here. Perhaps the additional explanation could just be something similar to "Expected the operand of the unary op to match the prescribed operand pattern".